doc/mkdocs/data/create_new_library.md

```diff
@@ -0,0 +1,644 @@
+
+#### Engine
+
+The engine is the framework that calls and executes the code implementing the Blocks. We currently provide support for Simulink and Simulink Coder. Alternative engines might be Scilab or even standalone C++ code.
+
+#### BlockInformation
+
+BlockInformation is the interface between a Block and the engine that executes its code. Despite blocks are not aware of what engine is executing them, blocks and engine need to exchange information such as user parameters and Port data. BlockInformation provides methods for setting and getting this piece of information.
+
+#### Simulink Block
+
+A Simulink Block is the wrapper of a (C++) Block. It provides a visual representation of it, with input and output ports and gives the possibility to connect it with other Simulink blocks through signals. The terms above come from this representation and map it to C++.
+
+#### Simulink Model
+
+A Simulink Model is a project composed of a set of Simulink Blocks interconnected by Signals.
+
+#### Simulink Library
+
+A Simulink Library is a container of a set of Blocks that populates the _Simulink Library Browser_. Every toolbox must be shipped with an associated Simulink Library file in order to expose its Blocks.
+
+#### S-Function
+
+There are many types of functions for implementing an algorithm wrapped by a Simulink Block. In Matlab terminology, these functions are referred as [S-Functions](https://it.mathworks.com/help/simulink/sfg/what-is-an-s-function.html). You can read more about the supported types in [What type of S-Function should you use](https://it.mathworks.com/help/simulink/sfg/what-type-of-s-function-should-you-use.html).
+
+In short S-Functions provide a sort of interface (through C callbacks) where a given number of functions need to be implemented. Simulink knows the name of these functions and calls them on demand.
+
+#### Block Mask
+
+A Simulink Block is just a square with input and output ports. It is possible to "mask" a Simulink Block in order to provide a user-friendly GUI that can be useful e.g. for setting Block parameters. The mask may contain buttons, dropdown menus, ...
+
+#### Shared library
+
+A shared library is an object that contains compiled code (functions, classes, ...) which cannot be executed standalone. They can be either _static_ or _dynamic_. Static shared libraries are meant to be embedded inside the executable that calls their symbols, instead the code of dynamic shared libraries is only referenced inside the executable and called when needed.
+
+For grasping better this difference, if multiple executables link against the same static library, the same code is embedded inside all of them, resulting in bigger executables and code duplication. A dynamic library object instead can be shared by multiple executables that need only to know their location in the filesystem and which symbols they provide.
```

Comment:
I always found "shared" used as a synonym of "dynamic", so I am a bit confused by the definition of  "static shared library". Furthermore also saying that "A library is an object" is a bit confusing, at least based on the typical definition of "object files" in linkers literature. Furthermore, it would be nice to point the interested reader to some resource on this, such as https://www.lurklurk.org/linkers/linkers.html .

Reply:
Yes, just _static library_ would suffice. Thanks for spotting it!